modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts

```diff
@@ -295,3 +282,93 @@ export async function encryptNShare(
     encryptedPrivateShare,
   };
 }
+
+export async function decryptNShare(encryptedNShare: DecryptableNShare): Promise<NShare> {
+  const privateShare = await readSignedMessage(
+    encryptedNShare.nShare.encryptedPrivateShare,
+    encryptedNShare.senderPublicArmor,
+    encryptedNShare.recipientPrivateArmor
+  );
+
+  const nShare: NShare = {
+    i: encryptedNShare.nShare.i,
+    j: encryptedNShare.nShare.j,
+    y: encryptedNShare.nShare.publicShare.slice(0, 65),
+    u: privateShare,
+    n: encryptedNShare.nShare.publicShare.slice(129),
+    chaincode: encryptedNShare.nShare.publicShare.slice(65, 129),
+  };
+
+  return nShare;
+}
+
+export function getPublicKey(commonKeyChain: string): string {
+  return commonKeyChain.slice(0, 66);
+}
+
+export function parseKShare(share: SignatureShareRecord): KShare {
```

Comment:
Lets add a condition for all these parsers to throw an error if it is not meeting the minimum length for the share